### PR TITLE
Provide option to default to github provider for git.api.getRemoteSources

### LINKED ISSUE #106316: Provide option to default to github provider git.api.getRemoteSources and potentially list org repos.
The repository picker feature https://github.com/microsoft/vscode/issues/102394 allows to pick github repos after selecting the GitHub provider. 
Asking for 
1. An option to default the provider to GitHub so that user can have one less click when creating a codespace. 
2. Currently it lists all repos where you are owner, is it possible to list repos of organizations you are in as well?

A side question is how does the api authenticate, is it with the stored git credentials in git extension? 

```
await commands.executeCommand('git.api.getRemoteSources', {
	providerLabel: (provider: RemoteSourceProvider) => `Test this for ${provider.name}`,
	urlLabel: `Test this for URL`,
        defaultProvider: 'Github'
});
```

@joaomoreno

Thanks,
Jason

### PR DESCRIPTION
**Pull request by wangjaso**
This PR fixes #106316 

@joaomoreno 

Add an optional param to the git.api.getRemoteSources api call to default to a specific provider.

You can test it by calling 
```
await commands.executeCommand('git.api.getRemoteSources', {
    providerName: 'GitHub'
})
```

This will skip the choose provider and show directly your repos in GitHub

### CODE REVIEW

File: extensions/git/src/remoteSource.ts [hunk 1]
```diff
@@ -80,54 +80,66 @@ class RemoteSourceProviderQuickPick {
...
-
-			if (remote) {
-				if (typeof remote.url === 'string') {
-					return remote.url;
-				} else if (remote.url.length > 0) {
-					return await window.showQuickPick(remote.url, { ignoreFocusOut: true, placeHolder: localize('pick url', "Choose a URL to clone from.") });
-				}
+	const targetedProvider = model.getRemoteProviders().filter(provider => provider.name === options.providerName);
+	if (targetedProvider && targetedProvider.length === 1) {
+		await pickProviderSource(targetedProvider[0]);
```
Comment (joaomoreno) [inline extensions/git/src/remoteSource.ts:92]: Isn't a `return` missing here?

**Review by joaomoreno — CHANGES_REQUESTED** [review]

**Comment by wangjaso** [comment]
Let me know if there's anything else to be fixed here. Thanks!

**Comment by wangjaso** [comment]
@joaomoreno Can this change be merged now?

**Comment by joaomoreno** [comment]
Thanks! 🍻

**Pull request merged**